### PR TITLE
refactor: small IQv2 cleanup

### PR DESCRIPTION
**Pull request by vvcephei**
### Description 
Just a few cleanup items after #8556 

### Testing done 
Running the CIT suite

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java
Comment (vvcephei) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java:76]: These are checking for conditions that are in the IQv2 contract not to happen. Of course, there could be upstream bugs, but I don't know if we really need to check for them in our code.

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java
Comment (vvcephei) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java:67]: Just a little inlining, since we only reference the map once, now.

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java
Comment (vvcephei) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java:71]: I initially suggested just calling `getResult` and letting the FailedResult class throw its own exception, but decided that's inappropriate, since that exception's message implies you shouldn't attempt to get the result of a failed query.

In any case, I thought we could give a little more context in the exception message, so I expanded on the message we throw.

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java
Comment (vvcephei) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java:82]: `if {throw/return} else if ...` is generally preferable to `if {throw/return} if ...` even through they're functionally equivalent because it makes the control flow clearer.

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java
Comment (vvcephei) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java:165]: Here's the more detailed message we're throwing now.

File: ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2Test.java
Comment (vvcephei) [inline ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2Test.java:130]: This was leftover from test debugging. I didn't want to hold up the other PR on this one fix, so I'm doing it after the merge.

**Review by agavra — APPROVED** [review]

**Pull request merged**
